hooks/useClickOutside.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import type { RefObject, MouseEvent, TouchEvent } from 'react';
+
+// eslint-disable-next-line no-unused-vars
+type TypeHandler = (event: MouseEvent | TouchEvent) => void;
+
+export const useClickOutside = (
+  ref: RefObject<HTMLElement>,
+  handler: TypeHandler
+) => {
+  useEffect(() => {
+    const listener = (event: MouseEvent | TouchEvent) => {
+      if (!ref.current || ref.current.contains(event.target as Node)) {
+        return;
+      }
+      handler(event);
+    };
+
+    document.addEventListener('mousedown', listener);
```

Comment:
Let's make these event listeners passive, just to make them non-blocking.

Reply:
just that ?
```js
document.addEventListener('mousedown', listener, { passive: true });
document.addEventListener('touchstart', listener, { passive: true });
```